ui/src/components/CheckboxTree/CheckboxSubTree.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import CheckboxRowWrapper from './CheckboxTreeRowWrapper';
+import { getCheckedCheckboxesCount } from './CheckboxTree.utils';
+import {
+    CheckboxContainer,
+    CheckboxWrapper,
+    Description,
+    GroupLabel,
+    RowContainer,
+    StyledCollapsiblePanel,
+} from './StyledComponent';
+import { GroupWithRows, ValueByField } from './types';
+
+interface CheckboxSubTreeProps {
+    group: GroupWithRows;
+    values: ValueByField;
+    handleRowChange: (newValue: { field: string; checkbox: boolean; text?: string }) => void;
+    disabled?: boolean;
+    handleParentCheckboxTree: (groupLabel: string, newCheckboxValue: boolean) => void;
+}
+
+const CheckboxSubTree: React.FC<CheckboxSubTreeProps> = ({
+    group,
+    values,
+    handleRowChange,
+    disabled,
+    handleParentCheckboxTree,
+}) => {
+    const [isExpanded, setIsExpanded] = useState(group.options?.expand);
+
+    const isParentChecked = useMemo(
+        () => group.rows.every((row) => values.get(row.field)?.checkbox),
+        [group.rows, values]
+    );
+
+    const isIndeterminate = useMemo(
+        () => group.rows.some((row) => !isParentChecked && values.get(row.field)?.checkbox),
+        [group.rows, values, isParentChecked]
+    );
+
+    const checkedCheckboxesCount = useMemo(
+        () => getCheckedCheckboxesCount(group, values),
+        [group, values]
+    );
+
+    useEffect(() => {
+        setIsExpanded(group.options?.expand);
+    }, [group.options?.expand, group.rows]);
+
+    const toggleCollapse = () => setIsExpanded((prev) => !prev);
+
+    const ParentCheckbox = (
+        <CheckboxWrapper>
+            <input
+                id={`checkbox-${group.label}`}
+                type="checkbox"
+                checked={isParentChecked}
+                ref={(el) => {
+                    const inputElement = el as HTMLInputElement | null;
+                    if (inputElement) {
+                        inputElement.indeterminate = isIndeterminate;
+                    }
+                }}
+                onChange={() => handleParentCheckboxTree(group.label, !isParentChecked)}
+                disabled={disabled}
+            />
+            <label htmlFor={`checkbox-${group.label}`}>{group.label}</label>
+        </CheckboxWrapper>
+    );
+
+    const childRows = (
+        <RowContainer>
+            {group.rows.map((row) => (
+                <CheckboxRowWrapper
+                    key={`row_${row.field}`}
+                    disabled={disabled}
+                    row={row}
+                    values={values}
+                    handleRowChange={handleRowChange}
+                />
+            ))}
+        </RowContainer>
+    );
+
+    const description = (
+        <Description>
+            {checkedCheckboxesCount} of {group.rows.length}
```

Comment:
It is a bit off
<img width="402" alt="image" src="https://github.com/user-attachments/assets/a471f566-342d-4454-ba1a-20647c61e7c8" />

What I expect:
<img width="377" alt="image" src="https://github.com/user-attachments/assets/04efdeb1-87bf-41d5-a361-e1f8ba4a5bab" />

Reply:
resolved